### PR TITLE
Class 06

### PR DESCRIPTION
**Pull request by ChristopherSClosser**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+v<!DOCTYPE html>
```
Comment (traceyr) [inline index.html:1]: You have a random V here

File: index.html [hunk 2]
```diff
@@ -56,9 +56,11 @@
   </body>
   <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
   <script src="https://cdnjs.cloudflare.com/ajax/libs/handlebars.js/4.0.6/handlebars.js"></script>
-  <script type="text/javascript" src="script/raw.js"></script>
+  <script type="text/javascript" src="data/raw.json"></script>
+  <!-- <script type="text/javascript" src="script/raw.js"></script> -->
```
Comment (traceyr) [inline index.html:60]: Remove any code you do not need!

File: script/projects.js [hunk 3]
```diff
@@ -16,18 +16,48 @@ Project.prototype.toHtml = function() {
...
+    var data = JSON.parse(localStorage.rawData);
+  } else {
+    //ajax call
+    ($.getJSON('data/raw.json').then(function(data){
+      localStorage.setItem('data', data);
+      localStorage.rawData = JSON.stringify(data);
+    }));
+  }
+  // Project.makeProjects();
+  data.sort(function(a,b) {
```
Comment (traceyr) [inline script/projects.js:31]: I am pretty sure we moved these into their own functions!

File: script/projects.js [hunk 4]
```diff
@@ -16,18 +16,48 @@ Project.prototype.toHtml = function() {
...
+    Project.all.push(new Project(projectObject));
+  });
+  Project.all.forEach(function(myNewProjectObject) {
+    console.log(myNewProjectObject.title);
+    $('#project-list').append('<li>' + myNewProjectObject.title + '</li><hr>').hide();
+    $('#projects').append(myNewProjectObject.toHtml());
+  });
+};
 
+//creates array of project objects in order newest first
```
Comment (traceyr) [inline script/projects.js:45]: Remove any code you do not need!

File: script/raw.js [hunk 5]
```diff
@@ -30,8 +30,15 @@ var projectDataSet = [
     thumbImg: ' src="imgs/choco_pizza.png"'
   },
   {
+    title: '<a href="https://christophersclosser.github.io/bus-mall/">Bus Mall</a>',
```
Comment (traceyr) [inline script/raw.js:33]: I would delete this file since we no longer need it.

File: script/projects.js [hunk 6]
```diff
@@ -16,18 +16,48 @@ Project.prototype.toHtml = function() {
...
-  $('#projects').append(myNewProjectObject.toHtml());
-
-});
+//looks for localStorage and/or saves to
+Project.fetchAll = function() {
+  if (localStorage.rawData) {
+    var data = JSON.parse(localStorage.rawData);
+  } else {
+    //ajax call
+    ($.getJSON('data/raw.json').then(function(data){
```
Comment (traceyr) [inline script/projects.js:25]: Your $.getJSON call does not need to be inside of parenthesis. I would recommend removing them

File: script/projects.js [hunk 7]
```diff
@@ -16,18 +16,48 @@ Project.prototype.toHtml = function() {
...
+      localStorage.setItem('data', data);
+      localStorage.rawData = JSON.stringify(data);
+    }));
```
Comment (traceyr) [inline script/projects.js:28]: Every .then needs a success and a failure function. Make sure to add that in.

File: script/projects.js [hunk 8]
```diff
@@ -16,18 +16,48 @@ Project.prototype.toHtml = function() {
...
-
-//makes list of projects in nav
-projects.forEach(function(myNewProjectObject) {
-  $('#project-list').append('<li>' + myNewProjectObject.title + '</li><hr>').hide();
-  $('#projects').append(myNewProjectObject.toHtml());
-
-});
+//looks for localStorage and/or saves to
+Project.fetchAll = function() {
+  if (localStorage.rawData) {
```
Comment (traceyr) [inline script/projects.js:21]: We worked on this together to make the code more modular.

File: script/projects.js [hunk 9]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-var projects = [];//this will store my projects
+Project.all = [];//this will store my projects
```
Comment (traceyr) [inline script/projects.js:3]: Great job moving this to make it part of the object!

File: script/projects.js [hunk 10]
```diff
@@ -16,18 +16,48 @@ Project.prototype.toHtml = function() {
...
+//     Project.all.push(new Project(projectObject));
+//   });
+//   Project.all.forEach(function(myNewProjectObject) {
+//     console.log(myNewProjectObject.title);
+//     $('#project-list').append('<li>' + myNewProjectObject.title + '</li><hr>').hide();
+//     $('#projects').append(myNewProjectObject.toHtml());
+//   });
+// });
+//displays correct project when selected
 var projectView = {};
```
Comment (traceyr) [inline script/projects.js:61]: Something to consider. All your code that is associated with projectView can be moved into its own projectView.js file. This organizes your code and makes it easier to find different parts!

File: index.html (hunk 1 above)
Comment (ChristopherSClosser) [reply index.html:1]: I was looking for that everywhere! Thanks

File: index.html (hunk 2 above)
Comment (ChristopherSClosser) [reply index.html:60]: I resubmitted before we worked on this

**Pull request merged**